helix-core/src/main/java/org/apache/helix/controller/rebalancer/waged/model/AssignableNode.java

```diff
@@ -138,15 +134,16 @@ void release(AssignableReplica assignableReplica) throws IllegalArgumentExceptio
           getInstanceName());
       return;
     }
-    Set<String> partitions = _currentAssignments.get(resourceName);
-    if (!partitions.contains(partitionName)) {
+    Map<String, String> partitionStateMap = _currentAssignments.get(resourceName);
+    if (!partitionStateMap.containsKey(partitionName) || !partitionStateMap.get(partitionName)
+        .equals(assignableReplica.getReplicaState())) {
```

Comment:
What if it's the second condition? If it's the second condition that's triggering this, then the log would be misleading.

Reply:
Will add "with state xxx" to the error message.